consent_manager/api.rst

```diff
@@ -0,0 +1,250 @@
+
+    The consent settings object.
+
+        Example::
+
+            {consents: {analytics: {status: -1}}}
+
+    Where `consent.analytics` is consent type and status indicate:
+
+    * `-1` - user does not interact, e.q. close consent popup without any decision
+    * `0` - user has rejected the consent
+    * `1` - user has approved the consent
+
+.. function:: onFulfilled()
+
+    The fulfilment handler callback. This function is **required**.
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code). If not specified, exception will be thrown in main stacktrace.
+
+    :param string|object error: **Required** Error code or exception.
+
+Get compliance settings
+`````````````````````````````````````
+Return current privacy settings. Might be useful for initializing custom decision view.
+When there is no decisions, just returns empty object. This state can be used to detect first time user interaction with consent mechanism.
+
+Code::
+
+    ppms.cm.api('getComplianceSettings', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getComplianceSettings', parameters: [onFulfilled, onRejected]});
+
+.. object:: settings
+
+    The consent settings object.
+
+        Example::
+
+            {consents: {analytics: {status: -1, updatedAt: '2018-07-03T12:18:19.957Z'}}}
+
+    Where `consent.analytics` is consent type and status indicate:
+
+    * `-1` - user has not interacted, e.g. has closed a consent popup without any decision
+    * `0` - user reject consent
+    * `1` - user approve consent
+
+.. function:: onFulfilled(settings)
+
+    The fulfilment handler callback (called with result). This function is **required**.
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code). If not specified, exception will be thrown in main stacktrace.
+
+    :param string|object error: **Required** Error code or exception.
+
+Send data request
+`````````````````````````````````````
+Command send data subject request to Consent Manager collector.
+
+Code::
+
+    ppms.cm.api('sendDataRequest', request, onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:sendDataRequest', parameters: [request, onFulfilled, onRejected]});
+
+.. object:: request
+
+    The subject data request.
+
+        Example::
+
+            {content: '', email: '', type: 'change_data|view_data|delete_data'}
+
+.. function:: onFulfilled()
+
+    The fulfilment handler callback. This function is **required**.
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code). If not specified, exception will be thrown in main stacktrace.
+
+    :param string|object error: **Required** Error code or exception.
+
+Example usage
+-------------
+Based on above listed commands there are many possibilities to implement custom consent gathering. Below is listed a simple implementation using jQuery library.
+
+.. literalinclude:: examples/api_full_example.html
```

Comment:
This example is quite long. I'm not sure how useful it'll be, but keeping it at the end of file shouldn't be too annoying.